otelcollector/configmapparser/tomlparser-scrape-interval.rb

```diff
@@ -0,0 +1,247 @@
+      else
+        @nodeexporterScrapeInterval = nodeexporterScrapeInterval
+        ConfigParseErrorLogger.log(LOGGING_PREFIX, "Using configmap scrape settings for nodeexporterScrapeInterval")
+      end
+    else
+      ConfigParseErrorLogger.logError(LOGGING_PREFIX, "nodeexporterScrapeInterval not specified")
+    end
+
+    windowsexporterScrapeInterval = parsedConfig[:windowsexporter]
+    if !windowsexporterScrapeInterval.nil?
+      matched = MATCHER.match(windowsexporterScrapeInterval)
+      if !matched
+        # set default scrape interval to 30s if its not in the proper format
+        windowsexporterScrapeInterval = "30s"
+        @windowsexporterScrapeInterval = windowsexporterScrapeInterval
+        ConfigParseErrorLogger.log(LOGGING_PREFIX, "Incorrect regex pattern for duration, set default scrape interval to 30s")
+      else
+        @windowsexporterScrapeInterval = windowsexporterScrapeInterval
+        ConfigParseErrorLogger.log(LOGGING_PREFIX, "Using configmap scrape settings for windowsexporterScrapeInterval")
+      end
+    else
+      ConfigParseErrorLogger.logError(LOGGING_PREFIX, "windowsexporterScrapeInterval not specified")
+    end
+
+    windowskubeproxyScrapeInterval = parsedConfig[:windowskubeproxy]
+    if !windowskubeproxyScrapeInterval.nil?
+      matched = MATCHER.match(windowskubeproxyScrapeInterval)
+      if !matched
+        # set default scrape interval to 30s if its not in the proper format
+        windowskubeproxyScrapeInterval = "30s"
+        @windowskubeproxyScrapeInterval = windowskubeproxyScrapeInterval
+        ConfigParseErrorLogger.log(LOGGING_PREFIX, "Incorrect regex pattern for duration, set default scrape interval to 30s")
+      else
+        @windowskubeproxyScrapeInterval = windowskubeproxyScrapeInterval
+        ConfigParseErrorLogger.log(LOGGING_PREFIX, "Using configmap scrape settings for windowskubeproxyScrapeInterval")
+      end
+    else
+      ConfigParseErrorLogger.logError(LOGGING_PREFIX, "windowskubeproxyScrapeInterval not specified")
+    end
+
+    prometheusCollectorHealthInterval = parsedConfig[:prometheuscollectorhealth]
+    if !prometheusCollectorHealthInterval.nil?
+      matched = MATCHER.match(prometheusCollectorHealthInterval)
+      if !matched
+        # set default scrape interval to 30s if its not in the proper format
+        prometheusCollectorHealthInterval = "30s"
+        @prometheusCollectorHealthInterval = prometheusCollectorHealthInterval
+        ConfigParseErrorLogger.log(LOGGING_PREFIX, "Incorrect regex pattern for duration, set default scrape interval to 30s")
+      else
+        @prometheusCollectorHealthInterval = prometheusCollectorHealthInterval
+        ConfigParseErrorLogger.log(LOGGING_PREFIX, "Using configmap scrape settings for prometheusCollectorHealthInterval")
+      end
+    else
+      ConfigParseErrorLogger.logError(LOGGING_PREFIX, "prometheusCollectorHealthInterval not specified")
```

Comment:
why is this an error ? it should eb info log, because we dont expect to have every target's scrape interval specified in config map. For ex;- out default config map doesn't have prom collector health scrape interval in the new settings section, so i see an error in the logs (see below). Rather just log an info saying no scrape interval overrides found in configmap for xyz target.

<img width="684" alt="image" src="https://user-images.githubusercontent.com/10353076/199168508-506895ec-5bf8-4c39-ad35-5c782584719e.png">

Reply:
Got it. Makes sense. Thanks Vishwa!